rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/validation/TrustAnchorValidationService.java

```diff
@@ -121,41 +123,41 @@ public void validate(long trustAnchorId) {
             return new TrustAnchorValidationRun(trustAnchorRef, trustAnchor.getLocations().get(0));
         });
 
+        final ValidationLocation trustAnchorValidationLocation = new ValidationLocation(validationRun.getTrustAnchorCertificateURI());
+        ValidationResult validationResult = ValidationResult.withLocation(trustAnchorValidationLocation);
+
         boolean updatedTrustAnchor = false;
+        final List<URI> rankedTrustAnchorLocations = trustAnchor.getLocationsByPreference();
         try {
+            boolean loadedAtLeastOneTACertificate = false;
```

Comment:
Can you hide this logic into a separate method? Here we basically only want to know if TA cert was fetched or not, without all the details.